### PR TITLE
Fixed problem #366 for networkservice/core/adapters. Added tests

### PR DESCRIPTION
**Pull request by ekanufrienko**
Signed-off-by: Ekaterina Anufrienko <ekanufrienko@mail.ru>

### CODE REVIEW

File: pkg/networkservice/core/adapters/capturecontext_server.go [hunk 1]
```diff
@@ -25,16 +25,21 @@ import (
...
+func (d *contextServer) Close(ctx context.Context, in *networkservice.Connection) (*empty.Empty, error) {
+	captureContext(ctx)
 	return next.Server(ctx).Close(ctx, in)
 }
 
-var _ networkservice.NetworkServiceServer = &doneServer{}
+// NewServer - creates a new networkservice.NetworkServiceServer chain element that adds
+// into context reference to current context for passing it from the adapter server/client to the next client/server
+// to avoid the problem with losing values from adapted server/client context.
+func NewServer() networkservice.NetworkServiceServer {
```
Comment (denis-tingajkin) [inline pkg/networkservice/core/adapters/capturecontext_server.go:43]: Please make this function private.

Comment (ekanufrienko) [reply pkg/networkservice/core/adapters/capturecontext_server.go]: ok, fixed

File: pkg/networkservice/core/adapters/capturecontext_client.go [hunk 2]
```diff
@@ -26,16 +26,21 @@ import (
...
 	return next.Client(ctx).Close(ctx, in, opts...)
 }
 
-var _ networkservice.NetworkServiceClient = &doneClient{}
+// newClient - creates a new networkservice.NetworkServiceClient chain element that adds
+// into context reference to current context for passing it from the adapter server/client to the next client/server
+// to avoid the problem with losing values from adapted server/client context.
+func newClient() networkservice.NetworkServiceClient {
+	return &contextClient{}
+}
```
Comment (denis-tingajkin) [inline pkg/networkservice/core/adapters/capturecontext_client.go:46]:
```suggestion
```

File: pkg/networkservice/core/adapters/client_to_server.go [hunk 3]
```diff
@@ -33,33 +33,35 @@ type clientToServer struct {
 
 // NewClientToServer - returns a networkservice.NetworkServiceServer wrapped around the supplied client
 func NewClientToServer(client networkservice.NetworkServiceClient) networkservice.NetworkServiceServer {
-	return &clientToServer{client: next.NewNetworkServiceClient(client, &doneClient{})}
+	return &clientToServer{client: next.NewNetworkServiceClient(client, newClient())}
```
Comment (denis-tingajkin) [inline pkg/networkservice/core/adapters/client_to_server.go:36]:
```suggestion
	return &clientToServer{client: next.NewNetworkServiceClient(client, &captureContextClient{})}
```

File: pkg/networkservice/core/adapters/capturecontext_client.go (hunk 2 above)
Comment (ekanufrienko) [reply pkg/networkservice/core/adapters/capturecontext_client.go:46]: ok, done

File: pkg/networkservice/core/adapters/client_to_server.go (hunk 3 above)
Comment (ekanufrienko) [reply pkg/networkservice/core/adapters/client_to_server.go]: ok, done

**Pull request closed without merging**